flash/core/data/data_source.py

```diff
@@ -543,3 +554,256 @@ def _get_classes(self, data):
+                            if isinstance(label, list):
+                                for item in label:
+                                    self.classes.add(item)
+                            else:
+                                self.classes.add(label)
+                            temp = {}
+                            temp["file_upload"] = task.get("file_upload")
+                            temp["data"] = task.get("data")
+                            temp["label"] = label
+                            temp["result"] = res.get("value")
+                            if annotation["ground_truth"]:
+                                self.test_results.append(temp)
+                            elif not annotation["ground_truth"]:
+                                self.results.append(temp)
+        self.num_classes = len(self.classes)
+        # splitting result to train and val sets
+        import random
+
+        random.shuffle(self.results)
+        data_length = len(self.results)
+        prop = data_length - int(data_length * self.split)
+        self.val_results = self.results[:prop]
+        self.results = self.results[prop:]
+
+    def load_sample(self, sample: Mapping[str, Any] = None, dataset: Optional[Any] = None) -> Any:
+        """Load 1 sample from dataset."""
+        if self._check_data_type("image"):
+            # extracting path to file
+            if sample["file_upload"]:
+                p = os.path.join(self._data_folder, sample["file_upload"])
+            else:
+                for key in sample.get("data"):
+                    p = sample.get("data").get(key)
+            # loading image
+            image = default_loader(p)
+            result = {
+                DefaultDataKeys.INPUT: image,
+                DefaultDataKeys.TARGET: self._get_labels_from_sample(sample["label"]),
+            }
+        elif self._check_data_type("text") and self.backbone:
+            data = ""
+            for key in sample.get("data"):
+                data += sample.get("data").get(key)
+            tokenized_data = self.tokenizer(data, max_length=self.max_length, truncation=True, padding="max_length")
+            for key in tokenized_data:
+                tokenized_data[key] = torch.tensor(tokenized_data[key])
+            tokenized_data["labels"] = self._get_labels_from_sample(sample["label"])
+            # separate text data type block
+            result = tokenized_data
+        elif self._check_data_type("video"):
+            result = sample
+        else:
+            # all other data types
+            input_data = deepcopy(sample)
+            try:
+                del input_data["label"]
+            except KeyError:
+                # no label in input data
+                pass
+            result = {
+                DefaultDataKeys.INPUT: input_data,
+                DefaultDataKeys.TARGET: self._get_labels_from_sample(sample["label"]),
+            }
+        return result
+
+    def generate_dataset(
```

Comment:
Ideally, we wouldn't override this and instead just return an `EncodedVideoDataset` in the `load_data` of a `LabelStudioVideoDataSource`